handlers/identity-backfill/src/main/scala/com/gu/identityBackfill/salesforce/GetSFContactSyncCheckFields.scala

```diff
@@ -54,14 +79,11 @@ object SyncableSFToIdentity {
   def apply(
     standardRecordType: RecordTypeId
   )(
-    fields: ContactSyncCheckFields
-  )(
-    sFContactId: SFContactId
-  ) =
-    for {
-      _ <- if (ContactSyncCheck(standardRecordType)(fields))
-        ContinueProcessing(())
-      else
-        ReturnWithResponse(ApiGatewayResponse.notFound("this sf contact can't be synced back to zuora/identity"))
-    } yield ()
+    fields: ContactsByAccountIdQueryResponse
```

Comment:
if we are following the philosophy of minimal coupling, if we are only using the "records" field we shouldn't pass in the whole query response object as the other fields are not necessary (this would add dependencies that weren't needed, which would harm testability)
in this case, better to pass in List[GetSFContactSyncCheckFields.ContactSyncCheckFields]
```suggestion
    fields: List[ContactSyncCheckFields]
```